languages/go/pkg/ffi.go

```diff
@@ -0,0 +1,251 @@
+package oso
```

Comment:
I would probably split this out as a separate `ffi` pkg. e.g. `github.com/osohq/oso/languages/go/pkg/ffi`. (Potentially `github.com/osohq/oso/languages/go/internal/pkg/ffi`)